src/test/java/org/opentripplanner/middleware/controllers/api/ManageLegTraversalTest.java

```diff
@@ -118,20 +126,125 @@ private static Stream<Arguments> createTrace() {
         );
     }
 
+    @ParameterizedTest
+    @MethodSource("createTurnByTurnTrace")
+    void canTrackTurnByTurn(
+        Coordinates position,
+        String expectedInstruction,
+        boolean isStartOfTrip,
+        String message
+    ) {
+        TravelerPosition travelerPosition = new TravelerPosition(
+            busStopToJusticeCenterItinerary.legs.get(0),
+            position
+        );
+        TripInstruction tripInstruction = alignTravelerToTrip(travelerPosition, isStartOfTrip);
+        if (tripInstruction != null) {
+            assertEquals(expectedInstruction, tripInstruction.build(), message);
+        } else {
+            assertEquals(expectedInstruction, NO_INSTRUCTION, message);
+        }
+    }
+
+    private static Stream<Arguments> createTurnByTurnTrace() {
+        List<Step> walkSteps = busStopToJusticeCenterItinerary.legs.get(0).steps;
+        Coordinates originCoords = new Coordinates(busStopToJusticeCenterItinerary.legs.get(0).from);
+        Step stepOne = walkSteps.get(0);
+        Coordinates stepOneCoords = new Coordinates(stepOne);
+        Step stepTwo = walkSteps.get(1);
+        Coordinates stepTwoCoords = new Coordinates(stepTwo);
+        Step stepThree = walkSteps.get(2);
+        Coordinates stepThreeCoords = new Coordinates(stepThree);
+        Step stepFour = walkSteps.get(3);
+        Coordinates stepFourCoords = new Coordinates(stepFour);
+        Step stepFive = walkSteps.get(4);
+        Coordinates stepFiveCoords = new Coordinates(stepFive);
```

Comment:
Can these things be an array or a list instead? It will be easier to build your arguments stream from that.

Reply:
I thought this was easier to follow by naming the steps and coords. The arguments don't always use the step with it's coords, e.g. "Approaching the destination." which is a combination of the last step and destination coords. Lemme know what you think, I may have misunderstood.